api/scraper/grades.js

```diff
@@ -21,11 +22,72 @@ var cheerio = require('cheerio');
 var cookie = require('cookie');
 var path = require('path');
 var unirest = require('unirest');
+var moment = require('moment');
 
 var status = require(path.join(__dirname, '..', 'status'));
 
 
 exports.get = function (app, data, callback) {
-    data.status = status.codes.toDo;
-    callback(false, data);
+      var gradesUri;
+      if (data.campus === 'vellore') {
+          gradesUri = 'https://academics.vit.ac.in/parent/student_history.asp';
+      }
+      else if (data.campus === 'chennai') {
+          gradesUri = 'http://27.251.102.132/parent/student_history.asp';
+      }
+      var CookieJar = unirest.jar();
+      var cookieSerial = cache.get(data.reg_no).cookie;
+      var onRequest = function (response) {
+          if (response.error) {
+              data.status = status.codes.vitDown;
+              console.log(JSON.stringify(data));
+              callback(true, data);
+          }
+          else {
+              var baseScraper = cheerio.load(response.body);
+              var coursesLength = baseScraper('table #hist tr').length;
+              var courses = [];
+              baseScraper('table #hist tr').each(function(i, elem) {
+                var attrs = baseScraper(this).children('td');
+                var course = {'course_code': attrs.eq(1).text(),
+                              'course_title': attrs.eq(2).text(),
+                              'course_type': attrs.eq(3).text(),
+                              'credits': parseInt(attrs.eq(4).text()),
+                              'grade': attrs.eq(5).text(),
+                              'exam_held': moment(attrs.eq(6).text()).format('YYYY-MM'),
+                              'result_date': ((attrs.eq(7).text() != "") ? (moment(attrs.eq(7).text()).format('YYYY-MM-DD')): null),
```

Comment:
@kishore-narendran: Instead of checking if it is a blank string, could you check using the isValid() function? Like how I'm doing it [here](https://github.com/aneesh-neelam/VITacademics/blob/development/api/scraper/marks.js#L152).